DEVELOPMENT.md

```diff
@@ -0,0 +1,213 @@
+# Recompiles the entire client before launching using `make dist/cata`
+make host
+
+# With file-watching so the server auto-restarts and recompiles on Go or TS changes:
+WATCH=1 make host
+
+# Delete all generated files (.pb.go and .ts proto files, and dist/)
+make clean
+
+# Recompiles the ts only for the given spec (e.g. make host_elemental_shaman)
+make host_$spec
+
+# Recompiles the `wowsimcata` server binary and runs it, hosting /dist directory at http://localhost:3333/cata.
+# This is the fastest way to iterate on core go simulator code so you don't have to wait for client rebuilds.
+# To rebuild client for a spec just do 'make $spec' and refresh browser.
+make rundevserver
+
+# With file-watching so the server auto-restarts and recompiles on Go or TS changes:
+WATCH=1 make rundevserver
+
+
+# The same as rundevserver, recompiles  `wowsimcata` binary and runs it on port 3333. Instead of serving content from the dist folder,
+# this command also runs `vite serve` to start the Vite dev server on port 5173 (or similar) and automatically reloads the page on .ts changes in less than a second.
+# This allows for more rapid development, with sub second reloads on TS changes. This combines the benefits of `WATCH=1 make rundevserver` and `WATCH=1 make host`
+# to create something that allows you to work in any part of the code with ease and speed.
+# This might get rolled into `WATCH=1 make rundevserver` at some point.
+WATCH=1 make devmode
+
+# This is just the same as rundevserver currently
+make devmode
+
+# This command recompiles the workers in the /ui/worker folder for easier debugging/development
+# Can be used with or without WATCH command
+make webworkers
+
+# With file watch enabled
+WATCH=1 make webworkers
+
+# Creates the 'wowsimcata' binary that can host the UI and run simulations natively (instead of with wasm).
+# Builds the UI and the compiles it into the binary so that you can host the sim as a server instead of wasm on the client.
+# It does this by first doing make dist/cata and then copying all those files to binary_dist/cata and loading all the files in that directory into its binary on compile.
+make wowsimcata
+
+# Using the --usefs flag will instead of hosting the client built into the binary, it will host whatever code is found in the /dist directory.
+# Use --wasm to host the client with the wasm simulator.
+# The server also disables all caching so that refreshes should pickup any changed files in dist/. The client will still call to the server to run simulations so you can iterate more quickly on client changes.
+# make dist/cata && ./wowsimcata --usefs would rebuild the whole client and host it. (you would have had to run `make devserver` to build the wowsimcata binary first.)
+./wowsimcata --usefs
+
+# Generate code for items. Only necessary if you changed the items generator.
+make items
+```
+
+# Adding a Sim
```

Comment:
These instructions are all out-of-date at this point but since they were already that way no need to fix them in this PR